doc/rtd/reference/datasources/nocloud.rst

```diff
@@ -4,97 +4,237 @@ NoCloud
 *******
 
 The data source ``NoCloud`` is a flexible datasource that can be used in
-multiple different ways. With NoCloud, one can provide configurations to
-the instance without running a network service (or even without having a
-network at all). Alternatively, one can use HTTP/HTTPS or FTP/FTPS to provide
-a configuration.
+multiple different ways.
 
-Configuration Methods:
+With NoCloud, one can provide configuration to the instance locally (without
+network access) or alternatively NoCloud can fetch the configuration from a
+remote server.
+
+Much of the following documentation describes how to tell cloud-init where
+to get its configuration.
+
+Runtime configurations
 ======================
 
-.. warning::
-    User data placed under ``/etc/cloud/`` will **not** be recognized as a
-    source of configuration data by the NoCloud datasource. While it may
-    be acted upon by cloud-init, using
-    :ref:`DataSourceNone<datasource_none_example>` should be preferred.
+Cloud-init discovers four types of configuration at runtime. The source of
+these configuration types is configurable with a meta-configuration. This
+meta-configuration can be delivered to cloud-init in different ways, but is
+different from the configurations that cloud-init uses to configure the
+instance at runtime.
+
+user-data
+---------
+
+User-data is a :ref:`configuration format<user_data_formats>` that allows a
```

Comment:
For all the sections here: https://cloudinit.readthedocs.io/en/latest/development/style_docs.html#common-words